services/provider/server/cephrbdmirror.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"context"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const rBDMirrorName = "rbd-mirror"
+
+type cephRBDMirrorManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newCephRBDMirrorManager(cl client.Client, namespace string) (*cephRBDMirrorManager, error) {
+	return &cephRBDMirrorManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (c *cephRBDMirrorManager) Create(ctx context.Context) error {
```

Comment:
any reason in not using creatorupdate from controller util?

Reply:
I was thinking about the case when we run MaintenanceMode (this requires the RBD mirror to be scaled down). If a new PeerBlockPool call is received while MaintenacneMode is in progress it and we use CreateOrUpdate, it will interfere with it.

Reply:
fair enough, does the owner scale up after the maintenance mode?

Reply:
Yes, that would be the responsibility of the controller executing the maintenance mode